pydm/widgets/channel.py

```diff
@@ -193,6 +193,10 @@ def disconnect(self, destroying=False):
         except Exception as exc:
             logger.exception("Unable to remove connection for %r", self)
         self._connected = False
+        if destroying:
+            return
+        if self.parent() and hasattr(self.parent(), 'connection_changed'):
```

Comment:
This could probably get reduced down to a one-line `if` statement but that is just me being pedantic

Reply:
My idea was to make it explicit that everything below the first if won't happen if we are destroying... maybe in the future we can add more items to be cleaned up on disconnection and if it was a one-liner it could be missed.